utils.sh

```diff
@@ -109,47 +114,56 @@ function network_ip() {
     echo "$ip"
 }
 
-function master_node_val() {
+function node_val() {
     local n
     local val
 
     n="$1"
     val="$2"
 
-    jq -r ".nodes[${n}].${val}" $MASTER_NODES_FILE
+    jq -r ".nodes[${n}].${val}" $NODES_FILE
 }
 
-function master_node_map_to_install_config() {
-    local num_masters
-    num_masters="$1"
+function node_map_to_install_config_hosts() {
+    local num_hosts
+    num_hosts="$1"
+    start_idx="$2"
+    role="$3"
 
-    for ((master_idx=0;master_idx<$1;master_idx++)); do
-      name=$(master_node_val ${master_idx} "name")
-      mac=$(master_node_val ${master_idx} "ports[0].address")
+    for ((idx=$start_idx;idx<$(($1 + $start_idx));idx++)); do
+      name=$(node_val ${idx} "name")
+      mac=$(node_val ${idx} "ports[0].address")
 
-      driver=$(master_node_val ${master_idx} "driver")
+      driver=$(node_val ${idx} "driver")
       if [ $driver == "ipmi" ] ; then
           driver_prefix=ipmi
       elif [ $driver == "idrac" ] ; then
           driver_prefix=drac
       fi
 
-      port=$(master_node_val ${master_idx} "driver_info.port // \"\"")
-      username=$(master_node_val ${master_idx} "driver_info.username")
-      password=$(master_node_val ${master_idx} "driver_info.password")
-      address=$(master_node_val ${master_idx} "driver_info.address")
+      port=$(node_val ${idx} "driver_info.port // \"\"")
+      username=$(node_val ${idx} "driver_info.username")
+      password=$(node_val ${idx} "driver_info.password")
+      address=$(node_val ${idx} "driver_info.address")
 
       cat << EOF
       - name: ${name}
-        role: master
+        role: ${role}
         bmc:
           address: ${address}
           username: ${username}
           password: ${password}
         bootMACAddress: ${mac}
-        hardwareProfile: ${MASTER_HARDWARE_PROFILE:-default}
 EOF
 
+        # FIXME(stbenjam) Worker code in installer should accept
+        # "default" as well -- currently the mapping doesn't work,
+        # so we use the raw value for BMO's default which is "unknown"
+        if [[ "$role" == "master" ]]; then
+          echo "        hardwareProfile: ${MASTER_HARDWARE_PROFILE:-default}"
+        else
+          echo "        hardwareProfile: ${WORKER_HARDWARE_PROFILE:-unknown}"
+        fi
```

Comment:
As a note for any other reviewers this will be fixed by https://github.com/openshift/installer/pull/2969